qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/BackupApiIT.java

```diff
@@ -0,0 +1,169 @@
+          S3.getSecretKey());
+  private final S3BackupStore store = new S3BackupStore(config);
+
+  private final ZeebeCluster cluster =
+      ZeebeCluster.builder()
+          .withImage(ZeebeTestContainerDefaults.defaultTestImage())
+          .withNetwork(NETWORK)
+          .withBrokersCount(2)
+          .withGatewaysCount(1)
+          .withReplicationFactor(1)
+          .withPartitionsCount(2)
+          .withEmbeddedGateway(false)
+          .withBrokerConfig(this::configureBroker)
+          .withNodeConfig(this::configureNode)
+          .build();
+
+  @RegisterExtension
+  @SuppressWarnings("unused")
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(cluster::getNodes);
+
+  @Container
+  private final ContainerEngine engine =
+      ContainerEngine.builder().withAutoAcknowledge(true).withCluster(cluster).build();
+
+  @AfterAll
+  static void afterAll() {
+    CloseHelper.quietCloseAll(NETWORK);
+  }
+
+  @BeforeEach
+  void beforeEach() {
+    try (final var s3Client = S3BackupStore.buildClient(config)) {
+      s3Client.createBucket(builder -> builder.bucket(config.bucketName()).build()).join();
+    }
+  }
+
+  @Test
+  void shouldTakeBackup() {
+    // given
+    final var actuator = BackupActuator.of(cluster.getAvailableGateway());
+    try (final var client = engine.createClient()) {
+      client.newPublishMessageCommand().messageName("name").correlationKey("key").send().join();
+    }
+
+    // when
+    final var response = actuator.take(1L);
+
+    // then
+    assertThat(response).isEqualTo(new TakeBackupResponse(1L));
+    Awaitility.await("until a backup exists with the given ID")
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(this::assertBackupCompleteOnAllPartitions);
+  }
+
+  private void assertBackupCompleteOnAllPartitions() {
+    // TODO: this will be replaced by the status API later
+    for (int partitionId = 1; partitionId < 2; partitionId++) {
+      assertBackupCompleteForPartition(partitionId);
+    }
+  }
+
+  private void assertBackupCompleteForPartition(final int partitionId) {
+    final var backupId = new BackupIdentifierImpl(0, partitionId, 1);
+    final var status = store.getStatus(backupId);
+
+    assertThat(status)
+        .succeedsWithin(Duration.ofSeconds(30))
+        .extracting(BackupStatus::id, BackupStatus::statusCode)
+        .containsExactly(backupId, BackupStatusCode.COMPLETED);
+  }
+
+  private void configureBroker(final ZeebeBrokerNode<?> broker) {
+    broker
+        .withEnv("ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEBACKUP", "true")
+        .withEnv("ZEEBE_BROKER_DATA_BACKUP_STORE", "S3")
+        .withEnv("ZEEBE_BROKER_DATA_BACKUP_S3_BUCKETNAME", config.bucketName())
+        // force using path-style instead of host-prefix, as otherwise the bucket will not be
+        // resolved
+        .withEnv(
+            "ZEEBE_BROKER_DATA_BACKUP_S3_ENDPOINT",
+            "http://%s:%d".formatted(getIpAddress(), S3_PORT))
```

Comment:
:wrench: How about 

```suggestion
            S3.getEndpointOverride(Service.S3)
```

`S3_PORT` and `getIpAddress` could be removed then.

Reply:
We need to use the IP address to enforce the S3 client to use a path-style approach instead of using the bucket name as a subdomain. If you use a domain name in general, it will use the bucket name as a subdomain. This is problematic in our setup since we have no real DNS resolver.

In fact, I was wondering if this is problem in general, even in production. Are all S3 compatible systems supporting this sub-domain based approach? I think the client can be configured to use path-style (e.g. `http://s3.local/myBucketName/route/etc` vs `http://myBucketName.s3.local/route/etc`), but I also read AWS wanted to phase out explicit path style in the future, so :shrug:

Reply:
This is what I was referring to in the comment, but I guess it wasn't clear that I needed the IP address explicitly for this.

Reply:
Oh, and you cannot use `getEndpointOverride` as this returns the externally accessible IP address (which is the Docker host IP) and not the internal container IP address (as both containers are on the same network).

Reply:
> Oh, and you cannot use getEndpointOverride as this returns the externally accessible IP address (which is the Docker host IP) and not the internal container IP address (as both containers are on the same network).

Ah, I missed that :+1: 

> If you use a domain name in general, it will use the bucket name as a subdomain. This is problematic in our setup since we have no real DNS resolver.

If S3 and Zeebe are in the same docker network, Zeebe should be able to resolve S3 over DNS though, right? Does it fail because the Docker DNS is not set up for subdomains?

> In fact, I was wondering if this is problem in general, even in production. Are all S3 compatible systems supporting this sub-domain based approach?

I'm not sure actually. MinIO can be configured to support it: https://github.com/minio/minio/tree/master/docs/config#domain.
Forcing users that can't use subdomains to set an IP address as endpoint is not ideal but we could extend the S3 store configuration to allow enabling/disabling path-style access.

Reply:
It didn't seem to be able to pick up the subdomain properly. If you give it an alias, e.g. `s3.local`, you could ping `s3.local`, but not `bucketname.s3.local`. I tried adding it as an extra host (such that `bucketname.s3.local` would resolve to the container IP), which worked, but then localstack didn't handle the request properly, so I'm guessing there's something missing with that. Possibly it's an issue with localstack itself? I didn't try with Minio.

Reply:
So according to localstack, you need to set the subdomain with the bucketname as a network alias as well. This is problematic if you want to reuse the container with a different bucket per test, since you don't know all the bucket names beforehand (unless you hardcode them).

I think we can use the IP for now as a workaround, I don't see too many downsides. The other option would be to use minio, which has a faster startup time, so we could start it on every test with the bucket name. I'm kind of neutral about it. I can timebox it, as it simplifies things to some extent.

Reply:
I don't mind either, do what you feel is best :+1:

Reply:
I rewrote it to use minio for 3 reasons:

1. It's closer to what we would expect someone to do on prod (e.g. use minio as an S3 alternative - not sure how close localstack is to s3)
2. Much faster startup
3. Easier to configure imo

Let me know what you think. And I already thought about reusing the `MinioContainer` in the backup module tests, but we can't without making that module dependent on `qa/util`, which doesn't seem right. We could push it to test-util I guess, but it also seems strange to have integration test utilities there. :shrug: